src/components/CodeChanges/CodeChanges.tsx

```diff
@@ -49,6 +52,36 @@ export const CodeChanges: React.FC = () => {
           (total, diff) => total + diff.deletions,
           0
         );
+
+        let codeChanges;
+
+        if (shouldPreserveCommits(fileDiffs)) {
+          codeChanges = bucketByCommit(fileDiffs).map((commitDiffs, idx) => {
+            const { description } = commitDiffs[0] ?? {};
+            const sortedFileDiffs = [...commitDiffs].sort((a, b) =>
+              a.fileName.localeCompare(b.fileName)
+            );
+            return (
+              <CodeChangeModuleContainer key={`code_change_${description}`}>
+                <CommitContainer>
+                  <CommitTitle>Commit {idx + 1}</CommitTitle>
+                  {description && (
+                    <CommitMessage data-cy="commit-name">
+                      {description}
+                    </CommitMessage>
+                  )}
+                </CommitContainer>
+                <CodeChangesTable fileDiffs={sortedFileDiffs} />
+              </CodeChangeModuleContainer>
+            );
+          });
+        } else {
+          const sortedFileDiffs = [...fileDiffs].sort((a, b) =>
```

Comment:
I couldn't change this one because `fileDiffs` is a `const` (immutable) and `sort` will try to sort the elements in place, which results in an error. So I have to make a copy by doing `[...fileDiffs]`